fdbclient/ReadYourWrites.actor.cpp

```diff
@@ -1770,7 +1770,10 @@ Future<int64_t> ReadYourWritesTransaction::getEstimatedRangeSizeBytes(const KeyR
 	if (resetPromise.isSet())
 		return resetPromise.getFuture().getError();
 
-	return map(waitOrError(tr.getDatabase()->getStorageMetrics(keys, -1), resetPromise.getFuture()),
+	// Pass in the TransactionState only if tenant is present
+	Optional<Reference<TransactionState>> trState =
+	    tr.trState->hasTenant() ? tr.trState : Optional<Reference<TransactionState>>();
```

Comment:
Nit: I think it would be okay to use a `Reference<TransactionState>` here, with an empty reference meaning there is no tenant info. Is the reason for using `Optional<Reference<TransactionState>>` to avoid accidental bad dereferencing?

Reply:
I was concerned about accidental bad deferencing, and wanted to make it clear in the interface for `getStorageMetrics()` that `TransactionState` may not always be available (and that when it is available, the behavior is different -- the function becomes tenant-aware). But I'm happy to just use `Reference<TransactionState>` to simplify the function signatures if you think that's better and stylistically more inline with FDB.

Reply:
I think `Reference<TransactionState>` is more in line with inline with FDB, but I can see the reasoning for wrapping it in `Optional`, so think this is okay.